File: PermCheck.cs
using System;
using System.Linq;

class Solution {
    public int solution(int[] A) {
        if (A.Length == 1 && A.First() != 1)
            return 0;
        
        Array.Sort(A);    
        
        if(A.First() != 1)
            return 0;
            
        bool isSequence = A.Zip(A.Skip(1), (current, next) => (current + 1) == next)
            .All(element => element);
            
        return isSequence ? 1: 0;
    }
}

File: perm_check.py
def solution(A):
    if len(A) == 1 and A[0] != 1:
        return 0
    
    A.sort()
    
    if A[0] != 1:
        return 0
        
    differences = [True if abs(current - next) == 1 else False for current, next in zip(A, A[1:])]
    return 1 if all(differences) else 0;